doc/mgr/modules.rst

```diff
@@ -48,13 +48,50 @@ or older versions of Ceph.
 Logging
 -------
 
-``MgrModule`` instances have a ``log`` property which is a logger instance that
-sends log messages into the Ceph logging layer where they will be recorded
-in the mgr daemon's log file.
+Logging in modules' code is done as in any other python program. Just import
+the ``logging`` package and get a logger instance with the
+``logging.getLogger`` function.
+
+Each module has a ``log_level`` option that specifies the current python
+logging level of the module.
+To change or query the logging level of the module use the following ceph
+commands::
+
+  ceph <module_name> logging level set <INFO|DEBUG|CRITICAL|ERROR|WARNING|NOTSET>
+  ceph <module_name> logging level get
+
+The logging level used upon the module's start is determined by the current
+logging level of the mgr daemon. The mgr daemon logging level is mapped to
+the module python logging level as follows:
+
+* <= 0 is CRITICAL
+* <= 1 is WARNING
+* <= 4 is INFO
+* <= +inf is DEBUG
+
+By default, modules' logging messages are processed by the Ceph logging layer
+where they will be recorded in the mgr daemon's log file.
+But it's also possible to send a module's logging message to it's own file.
+
+The module's log file will be located in the same directory as the mgr daemon's
+log file with the following name pattern::
+
+   <mgr_daemon_log_file_name>.<module_name>.log
+
+To enable the file logging on a module use the following command::
+
+   ceph <module_name> logging file enable
+
+When the module's file logging is enabled, module's logging messages stop
```

Comment:
It'd be great to add/specify the behaviour of the logger regarding the `debug_mgr` level overrides and the defaults.

Let's say a production operator wants to minimize log writes: do they need to reduce log levels in all potential module logs, or is it enough to reduce `debug_mgr` level?

On the contrary, I have a minimal mgr debug level, but want to increase a single module logger verbosity. Is that possible?

I'm just curious about the above 2 scenarios, as I can think of real use cases for them (probably they're contradictory, and only one can be implemented safely).

Reply:
I say in the docs this:
```
The logging level used upon the module's start is determined by the current
logging level of the mgr daemon.
```

If I change the above to:
```
The logging level used upon the module's start is determined by the current
logging level of the mgr daemon, unless if the ``log_level`` option was
previously set with the ``config set ...`` command.
```
does it make it more clear the behavior of the module logging with respect to the debug_mgr level?